library/ssl_client.c

```diff
@@ -714,6 +714,50 @@ static int ssl_generate_random( mbedtls_ssl_context *ssl )
     return( ret );
 }
 
+static int ssl_session_set_hostname( mbedtls_ssl_session *session,
```

Comment:
```suggestion
static int mbedtls_ssl_session_set_hostname( mbedtls_ssl_session *session,
```
declared in ssl_misc.h and defined in ssl_tls.c as we need to use it here and in ssl_tls.c (see comment in mbedtls_ssl_session_copy()).

Otherwise, an "#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)" guard is probably missing and we should probably also factorize the code of this function and that of mbedtls_ssl_set_hostname() in ssl_tls.c.